src/main/java/com/hazelcast/aws/AwsConfig.java

```diff
@@ -30,8 +30,7 @@
     private final String region;
     private final String hostHeader;
     private final String securityGroupName;
-    private final String tagKey;
-    private final String tagValue;
+    private final Tags tags;
```

Comment:
I think that having a separate class `Tags` does not simplify anything here. I guess having here `List<Tag>` would be cleaner. And the class `Tag` can be either a separate class or an inner class of `AwsConfig`.